apps/server/src/modules/task/entity/task.entity.ts

```diff
@@ -1,14 +1,16 @@
-import { Entity, ManyToOne, Property } from '@mikro-orm/core';
-import { ObjectId } from '@mikro-orm/mongodb';
+import { Collection, Entity, ManyToOne, OneToMany, Property } from '@mikro-orm/core';
 import { BaseEntityWithTimestamps, EntityId } from '@shared/domain';
+import { Course } from '@src/entities';
+import type { Submission } from './submission.entity';
 import { LessonTaskInfo } from './lesson-task-info.entity';
 
 interface ITaskProperties {
 	name: string;
 	dueDate?: Date;
 	private?: boolean;
-	parentId: ObjectId;
+	parent?: Course;
```

Comment:
really no

Reply:
well we could use a union type which includes all entities that can be associated as a parent => polymorphic association.

Reply:
where are id's part of a domain model? A parent can be a course and here it is. How to create logic when having an un-typed parent @CeEv ?

Reply:
Not untyped we have a interface for it and this should replaced the ObjectIds.

Reply:
but the parentId property is still based on an ObjectId right? so if that id is still pointing to another entity: what is the type of that target entity then? this should be clear in terms of type safety. if the target entity implements an interface that's ok. but it has to be clear how the actual entities are related to each other.
the information what the type of the target entity is has to be stored together with the id. only if we specifically define the possible types of the relation we can model our entities in a typesafe way.

Reply:
But you want to use it to populate the informations, if i see it right. This is different. Typesafe for a objectId sound confusing. The objectId is typesave it is an objectId. The parent is typesave because it has a defined interface. 
Without populate on repo layer the reference bring us no more ts informations an no other releated information.

Reply:
But shouldn't we even without populate know which type of entity the ObjectId references?